src/charm.py

```diff
@@ -67,7 +67,13 @@ def _on_remove(self, event):
     def _on_upgrade_action(self, event):
         """Upgrade the license-manager-cli package."""
         version = event.params["version"]
-        self._license_manager_cli_ops.upgrade(version)
+        try:
+            self._license_manager_cli_ops.upgrade(version)
+            event.set_results({"upgrade": "success"})
+        except Exception:
+            self.unit.status = BlockedStatus("Error upgrading cluster-agent")
+            event.fail(message="Error upgrading cluster-agent")
```

Comment:
```suggestion
            event.fail(message="Error upgrading license-manager-cli")
```

Reply:
fixed in [c175561](https://github.com/omnivector-solutions/charm-license-manager-cli/pull/2/commits/c1755612944964b362a0f62f7b2dcd8784672917)